lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -124,6 +124,22 @@ namespace Js
         JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedRegExp, propertyName);
     }
 
+    JavascriptString* JavascriptRegExp::GetFirstStringArg(Arguments& args, ScriptContext* scriptContext)
+    {
+        if (args.Info.Count == 1)
+        {
+            return scriptContext->GetLibrary()->GetUndefinedDisplayString();
+        }
+        else if (JavascriptString::Is(args[1]))
+        {
+            return JavascriptString::FromVar(args[1]);
```

Comment:
I think you don't need the second if check - you can simply pass to JavascriptConversion::ToString() and it will take care of that.